compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -3587,9 +3594,21 @@ void OMR::ValuePropagation::transformArrayCloneCall(TR::TreeTop *callTree, TR_Op
       }
    else
       {
-      TR::Node *loadaddr = TR::Node::createWithSymRef(callNode, TR::loadaddr, 0, comp()->getSymRefTab()->findOrCreateClassSymbol(callNode->getSymbolReference()->getOwningMethodSymbol(comp()), 0, j9class));
+      TR::Node *classNode;
+      if (isFixedClass)
+         {
+         classNode = TR::Node::createWithSymRef(callNode, TR::loadaddr, 0, comp()->getSymRefTab()->findOrCreateClassSymbol(callNode->getSymbolReference()->getOwningMethodSymbol(comp()), 0, j9class));
+         }
+      else
+         {
+         // Load the component class of the cloned array as 2nd child of anewarray as expected by the opcode.
+         TR::Node * arrayClassNode = TR::Node::createWithSymRef(callNode, TR::aloadi, 1, objNode, comp()->getSymRefTab()->findOrCreateVftSymbolRef());
+         classNode = TR::Node::createWithSymRef(callNode, TR::aloadi, 1, arrayClassNode, comp()->getSymRefTab()->findOrCreateArrayComponentTypeSymbolRef());
+         }
       TR::SymbolReference *symRef = comp()->getSymRefTab()->findOrCreateANewArraySymbolRef(objNode->getSymbolReference()->getOwningMethodSymbol(comp()));
-      TR::Node::recreateWithoutProperties(callNode, TR::anewarray, 2, lenNode, loadaddr, symRef);
+      TR::Node::recreateWithoutProperties(callNode, TR::anewarray, 2, lenNode, classNode, symRef);
```

Comment:
I was concerned about the behaviour of `anewarray` given a primitive as the component type (exception? UB?). But in order to reach this point, the cloned object must have been constrained to be a reference array. If the object could be either a reference array or a primitive array, then the type bound would not have been an array type, and `isArray()` would have been `TR_maybe`. So for the purpose of this change the primitive `anewarray` question is moot :+1: